libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtilsTest.kt

```diff
@@ -2063,17 +2084,216 @@ class MapboxRouteLineUtilsTest {
+            every {
+                directionsRoute.legs()!![0].annotation()
+            } returns route.directionsRoute.legs()!![0].annotation()
+            every { directionsRoute.routeOptions() } returns route.directionsRoute.routeOptions()
+            every {
+                directionsRoute.legs()!![0].annotation()!!.congestionNumeric()
+            } returns route.directionsRoute
+                .legs()!![0]
+                .annotation()!!
+                .congestionNumeric()!!
+                .shuffled()
+        }
+        val trafficProvider =
+            MapboxRouteLineUtils.getRouteLegTrafficNumericCongestionProvider(colorResources)
+        val eqKey = CacheResultUtils.CacheResultKeyRouteTraffic<Any>(route2, trafficProvider)
+
+        val equals =
+            CacheResultUtils.CacheResultKeyRouteTraffic<Any>(route, trafficProvider).equals(eqKey)
+
+        assertFalse(equals)
+    }
+
+    @Test
+    fun cacheResultKeyRouteTraffic_closuresNotEqual() {
+        val trafficProvider = MapboxRouteLineUtils.getRouteLegTrafficCongestionProvider
+        val route = loadNavigationRoute("route-with-closure.json")
+        val route2 = mockk<NavigationRoute> {
+            every { id } returns route.id
+            every { directionsRoute } returns route.directionsRoute
+            every { directionsRoute.legs() } returns route.directionsRoute.legs()
+            every { directionsRoute.legs()!!.size } returns route.directionsRoute.legs()!!.size
+            every {
+                directionsRoute.legs()!![0].annotation()
+            } returns route.directionsRoute.legs()!![0].annotation()
+            every { directionsRoute.routeOptions() } returns route.directionsRoute.routeOptions()
+            every { directionsRoute.legs()!![0].closures() } returns listOf(
+                route.directionsRoute
+                    .legs()!!
+                    .first()
+                    .closures()!!
+                    .first()
+                    .toBuilder()
+                    .geometryIndexStart(5)
+                    .build()
+            )
         }
+        val eqKey = CacheResultUtils.CacheResultKeyRouteTraffic<Any>(route2, trafficProvider)
+
+        val equals =
+            CacheResultUtils.CacheResultKeyRouteTraffic<Any>(route, trafficProvider).equals(eqKey)
+
+        assertFalse(equals)
+    }
+
+    @Test
+    fun cacheResultKeyRouteTraffic_closuresEqual() {
+        val trafficProvider = MapboxRouteLineUtils.getRouteLegTrafficCongestionProvider
+        val route = loadNavigationRoute("route-with-closure.json")
+        val eqKey = CacheResultUtils.CacheResultKeyRouteTraffic<Any>(route, trafficProvider)
+
+        val equals =
+            CacheResultUtils.CacheResultKeyRouteTraffic<Any>(route, trafficProvider).equals(eqKey)
 
-        return first.zip(second).all { (x, y) ->
-            equalityFun(x, y)
+        assertTrue(equals)
+    }
+
+    @Test
+    fun cacheResultKeyRouteTraffic_roadClassesNotEqual() {
```

Comment:
You can also check:
1) different road classes size.

Reply:
If the sizes are different CacheResultUtils::listElementsAreEqual will return false.

Reply:
Yes, but as far as I can see, there are no tests covering that.